compiler/noirc_evaluator/src/ssa/ir/instruction/call.rs

```diff
@@ -233,7 +233,20 @@ pub(super) fn simplify_call(
                 SimplifyResult::None
             }
         }
-        Intrinsic::ApplyRangeConstraint => SimplifyResult::None,
+        Intrinsic::ApplyRangeConstraint => {
+            let value = arguments[0];
+            let max_bit_size = dfg.get_numeric_constant(arguments[1]);
```

Comment:
shouldn't this always be a numeric constant, given that we call assert_constant on the bitsize?

Reply:
```
Initial SSA:
acir fn main f0 {
  b0(v0: Field):
    call f1(v0, u32 2⁴×3)
    return 
}
acir fn assert_maximum_bit_size f1 {
  b0(v0: Field, v1: u32):
    call assert_constant(v1)
    v5 = lt v1, u32 254
    constrain v5 == u1 1
    call apply_range_constraint(v0, v1)
    return 
}
```
That requires inlining to happen first.